Code/Tools/AssetProcessor/native/utilities/ProductOutputUtil.cpp

```diff
@@ -0,0 +1,147 @@
+                    AZStd::string newName = (AZ::IO::FixedMaxPath(AZ::IO::PathView(product.m_productFileName).ParentPath()) /
+                                             unprefixedString.toUtf8().constData())
+                                                .AsPosix()
+                                                .c_str();
+
+                    AssetUtilities::ProductPath oldProductPath(product.m_productFileName, platformIdentifier);
+                    AssetUtilities::ProductPath newProductPath(newName, platformIdentifier);
+                    AssetProcessor::ProductAssetWrapper wrapper{ product, oldProductPath };
+
+                    auto oldAbsolutePath = wrapper.HasCacheProduct() ? oldProductPath.GetCachePath() : oldProductPath.GetIntermediatePath();
+                    auto newAbsolutePath = wrapper.HasCacheProduct() ? newProductPath.GetCachePath() : newProductPath.GetIntermediatePath();
+
+                    product.m_productFileName = newName;
+
+                    // Find any other sources which output the non-prefixed product
+                    AzToolsFramework::AssetDatabase::ProductDatabaseEntryContainer existingProducts;
+                    if (db->GetProductsByProductName(newProductPath.GetDatabasePath().c_str(), existingProducts))
+                    {
+                        for (const auto& existingProduct : existingProducts)
+                        {
+                            AzToolsFramework::AssetDatabase::SourceDatabaseEntry existingSource;
+                            if (db->GetSourceByProductID(existingProduct.m_productID, existingSource))
+                            {
+                                if (existingSource.m_scanFolderPK != sourceAsset.ScanFolderId() ||
+                                    AZ::IO::PathView(existingSource.m_sourceName) != sourceAsset.RelativePath())
+                                {
+                                    // Found a different source already using this product name
+                                    // This should be a previously-higher priority override
+                                    // Rename the existing file/product entries from non-prefixed to prefixed version
+                                    RenameProduct(db, existingProduct, existingSource);
+                                }
+                            }
+                        }
+                    }
+
+                    bool result = AssetUtilities::MoveFileWithTimeout(oldAbsolutePath.c_str(), newAbsolutePath.c_str(), 1);
+
+                    if (!result)
+                    {
+                        AZ_Error(
+                            "ProductOutputUtil",
+                            false,
+                            "Failed to move product from " AZ_STRING_FORMAT " to " AZ_STRING_FORMAT,
+                            AZ_STRING_ARG(oldAbsolutePath),
+                            AZ_STRING_ARG(newAbsolutePath));
```

Comment:
This call doesn't store data on the specific failure anywhere you can look up, but it does print out log messages that can help track why. It would probably be worth updating the error message below to have at the end "See previous log messages for details on why the move failed.", so people skimming or just looking at error output will know to look at the non-error output.

Same with farther down in RenameProduct.